src/plugins/inputs/join.js

```diff
@@ -0,0 +1,33 @@
+"use strict";
+
+var Chan = require("../../models/chan");
+var _ = require("lodash");
+
+exports.commands = ["join"];
+
+exports.input = function(network, chan, cmd, args) {
+	var client = this;
+	if (args.length !== 0) {
+		var irc = network.irc;
+		irc.join(args[0], args[1]);
+	}
+
+	_.zip(args[0].split(/,/g), args[1].split(/,/g))
+		.map(function(chanPairs) {
+			return new Chan({
+				type: Chan.Type.CHANNEL,
+				name: chanPairs[0],
+				key: chanPairs[1],
+			});
+		})
+		.map(function(channel) {
+			network.channels.push(channel);
+			client.emit("join", {
+				network: network.id,
+				chan: channel
+			});
+		});
+
+	this.save();
```

Comment:
You know, when I was testing, the channels keys weren't saved in the user JSON files when I didn't add this line. I could always try again.